### PR TITLE
[AutoDiff] Fix AD diagnostics tracking.

### PR DESCRIPTION
**Pull request by rxwei**
Fix DifferentiationInvoker so that errors are guaranteed to be reported in the differentiation pass.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/TFDifferentiation.cpp [hunk 1]
```diff
@@ -3439,7 +3464,7 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
     // TODO: Change this to a note when we emit an error at the @autodiff
```
Comment (dan-zheng) [inline lib/SILOptimizer/Mandatory/TFDifferentiation.cpp]: Can this TODO note be removed?

**Review by dan-zheng — APPROVED** [review]
Hurray for fixed diagnostics!

File: lib/SILOptimizer/Mandatory/TFDifferentiation.cpp (hunk 1 above)
Comment (rxwei) [reply lib/SILOptimizer/Mandatory/TFDifferentiation.cpp]: Good catch! I think I’ll get rid of em in my next PR since CI has passed.

**Pull request merged**